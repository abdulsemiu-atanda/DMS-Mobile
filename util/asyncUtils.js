import sendRequest from '../requests/requestUtil'

/**
 * Constructs async actions constants
 * @param {string} baseName
 * @returns {Object}
 */
export const asyncActionNames = baseName => ({
  failure: `${baseName}_FAILURE`,
  loading: `${baseName}_LOADING`,
  success: `${baseName}_SUCCESS`
})

/**
 * Constructs Redux async actions
 * @param {string} actionName
 * @returns {Object}
 */
export const asyncActions = (actionName) => ({
  loading: bool => ({
    type: asyncActionNames(actionName).loading,
    data: bool
  }),
  failure: (bool, error) => ({
    type: asyncActionNames(actionName).failure,
    data: {error, status: bool}
  }),
  success: data => ({
    type: asyncActionNames(actionName).success,
    data
  })
})

/**
 * Dispatches Redux Actions
 * @async asyncRequest
 * @param {string} ACTION_NAME - The name of action to be dispatched
 * @param {string} endpoint - The endpoint to resource
 * @param {string} method - HTTP method
 * @param {Object} data - HTTP request body
 * @param {string} token - Authentication token for request
 * @returns {Function} - Function that dispatches Redux Actions
 */

export const asyncRequest = (ACTION_NAME, endpoint, method, data, token) => dispatch => {
  dispatch(asyncActions(ACTION_NAME).loading(true))
  sendRequest(endpoint, method, data, token)
    .then(res => res.json())
    .then(res => {
      dispatch(asyncActions(ACTION_NAME).success(res))
      dispatch(asyncActions(ACTION_NAME).loading(false))
    })
    .catch(err => dispatch(asyncActions(ACTION_NAME).failure(true, err)))
}
