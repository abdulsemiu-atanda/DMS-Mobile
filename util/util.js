import {
  emailCorrect,
  emailError,
  fullnameCorrect,
  fullnameError,
  passwordMatch,
  passwordMatchError,
  usernameCorrect,
  usernameError
} from '../actions/formActions'

export const validateFields = data => {
  return dispatch => {
    Object.keys(data).forEach(field => {
      switch (field) {
        case 'email':
          return /^\S+@\S+$/g.test(data[field]) ? dispatch(emailCorrect(true)) : dispatch(emailError(true))
        case 'username':
          return data[field].trim() === '' ? dispatch(usernameError(true)) : dispatch(usernameCorrect(true))
        case 'password':
          return data.password === data.confirmPassword ? dispatch(passwordMatch(true)) : dispatch(passwordMatchError(true))
        case 'fullname':
          return data[field].trim() === '' ? dispatch(fullnameError(true)) : dispatch(fullnameCorrect(true))
        default:
          break
      }
    })
  }
}