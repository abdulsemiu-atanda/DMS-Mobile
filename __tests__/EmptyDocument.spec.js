import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import EmptyDocument from '../components/shared/EmptyDocument.react'
import {noop} from '../util/util'

const props = {
  screen: 'All',
  addDocument: noop
}

describe('EmptyDocument', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<EmptyDocument {...props} />).toJSON()

    expect(tree).toMatchSnapshot()
    expect(tree.children[1].children[0])
      .toMatch('Oops! there are no public documents currently available for viewing.')
  })

  it('renders correctly when screen is not All', () => {
    const newProps = {
      ...props,
      screen: 'Collection'
    }
    const tree = renderer.create(<EmptyDocument {...newProps} />).toJSON()

    expect(tree).toMatchSnapshot()
    expect(tree.children[1].children[0]).toEqual('Oops! You have not added any document.')
  })
})
