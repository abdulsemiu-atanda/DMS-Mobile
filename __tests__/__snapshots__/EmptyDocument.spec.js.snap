// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyDocument renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "grey",
          "fontSize": 70,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "Verdana",
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
  >
    Oops! there are no public documents currently available for viewing.        Please check again later or start by adding a public document.
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignContent": "center",
        "alignSelf": "center",
        "backgroundColor": "#0983ff",
        "padding": 7,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#aacfff",
          "fontSize": 15,
          "textAlign": "center",
        }
      }
    >
      ADD DOCUMENT
    </Text>
  </View>
</View>
`;

exports[`EmptyDocument renders correctly when screen is not All 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "grey",
          "fontSize": 70,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "Verdana",
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
  >
    Oops! You have not added any document.
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignContent": "center",
        "alignSelf": "center",
        "backgroundColor": "#0983ff",
        "padding": 7,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#aacfff",
          "fontSize": 15,
          "textAlign": "center",
        }
      }
    >
      ADD DOCUMENT
    </Text>
  </View>
</View>
`;
